src/utils/sample/discrete_gauss.rs

```diff
@@ -115,6 +127,88 @@ fn gaussian_function(x: &Z, c: &Q, s: &Q) -> Q {
     res.exp_taylor(100u32)
 }
 
+/// SampleD samples a discrete Gaussian from the lattice with `basis` using [`sample_z`] as a subroutine.
+///
+/// Parameters:
+/// - `basis`: specifies a basis for a lattice
+/// - `n`: specifies the range from which [`sample_z`] samples
+/// - `center`: specifies the positions of the center with peak probability
+/// - `s`: specifies the Gaussian parameter, which is proportional
+/// to the standard deviation `sigma * sqrt(2 * pi) = s`
+///
+/// Returns a vector with discrete gaussian error based on a lattice point
+/// as in [\[1\]](<index.html#:~:text=[1]>): SampleD.
+///
+/// # Example
+/// ```compile_fail
+/// use qfall_math::{integer::{MatZ, Z}, rational::{MatQ, Q}};
+/// use qfall_math::utils::sample::discrete_gauss::sample_d;
+/// let basis = MatZ::identity(5, 5).unwrap();
+/// let n = Z::from(1024);
+/// let center = MatQ::new(5, 1).unwrap();
+/// let gaussian_parameter = Q::ONE;
+///
+/// let sample = sample_d(basis, &n, &center, &gaussian_parameter).unwrap();
+/// ```
+///
+/// # Errors and Failures
+/// - Returns a [`MathError`] of type [`InvalidIntegerInput`](MathError::InvalidIntegerInput)
+/// if the `n <= 1` or `s <= 0`.
+/// - Returns a [`MathError`] of type [`MismatchingMatrixDimension`](MathError::MismatchingMatrixDimension)
+/// if the number of rows of the `basis` and `center` differ.
+/// - Returns a [`MathError`] of type [`InvalidMatrix`](MathError::InvalidMatrix)
+/// if `center` is not a row vector.
```

Comment:
The reference for the implementation is only included in the upper function, but imo, it should also be visible here